shared/modules/ibc_host_module.go

```diff
@@ -0,0 +1,197 @@
+package modules
+
+//go:generate mockgen -destination=./mocks/ibc_host_module_mock.go github.com/pokt-network/pocket/shared/modules IBCHostModule,IBCHandler
+
+const IBCHostModuleName = "ibc_host"
+
+type IBCHostOption func(IBCHostModule)
+
+type IBCHostFactory = FactoryWithOptions[IBCHostModule, IBCHostOption]
+
+// IBCHost is the interface used by the host machine (a Pocket node) to interact with the IBC module
+// the host is responsible for managing the IBC state and interacting with consensus in order for
+// any IBC packets to be sent to another host on a different chain (via an IBC relayer). The hosts
+// are also responsible for receiving any IBC packets from another chain and verifying them through
+// the light clients they manage
+// https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements
+type IBCHostModule interface {
```

Comment:
How would you feel about `IBCHostSubmodule` instead?

I'm thinking back to the discussion we had earlier about naming the implementation's file `submodule.go` over `module.go`. I acknowledge that here w're in the `modules` package but I would argue this is for 2 reasons, to avoid import cycles and reduce complexity.

I imagine the number of submodules eventually, if not quickly, outnumbering modules. I anticipate the effects of this will include cluttering up the bus module interface with getter functions and accumulating more interface files in this package. By differentiating the naming convention of submodule interfaces it should become much easier to make the distinction between them just from the name